cdk/src/constructs/enable-controltower.ts

```diff
@@ -0,0 +1,108 @@
+import * as path from 'path';
+import * as pythonLambda from '@aws-cdk/aws-lambda-python-alpha';
+import { aws_lambda as lambda, CfnCustomResource, CustomResource, Stack } from 'aws-cdk-lib';
+import * as iam from 'aws-cdk-lib/aws-iam';
+import * as cr from 'aws-cdk-lib/custom-resources';
+import { Construct } from 'constructs';
+
+
+interface EnableControltowerProps {
+  /**
+   * Email of the audit aws account
+   */
+  readonly auditAwsAccountEmail: string;
+  /**
+   * Email of the log archive aws account
+   */
+  readonly logArchiveAwsAccountEmail: string;
+}
+
+export class EnableControltower extends Construct {
+  constructor(scope: Construct, id: string, props: EnableControltowerProps) {
+    super(scope, id);
+
+    const resource = new CustomResource(this, 'Resource', {
+      serviceToken: EnableControltowerProvider.getOrCreate(this),
+      resourceType: 'Custom::EnableControltower',
+      properties: {
+        LOG_ARCHIVE_AWS_ACCOUNT_EMAIL: props.logArchiveAwsAccountEmail,
+        AUDIT_AWS_ACCOUNT_EMAIL: props.auditAwsAccountEmail,
+      },
+    });
+    (resource.node.defaultChild as CfnCustomResource).overrideLogicalId(id)
+  }
+}
+
+class EnableControltowerProvider extends Construct {
+
+  /**
+   * Returns the singleton provider.
+   */
+  public static getOrCreate(scope: Construct) {
+    const stack = Stack.of(scope);
+    const id = 'superwerker.enable-controltower';
+    const x = stack.node.tryFindChild(id) as EnableControltowerProvider || new EnableControltowerProvider(stack, id);
+    return x.provider.serviceToken;
+  }
+
+  private readonly provider: cr.Provider;
+
+  constructor(scope: Construct, id: string) {
```

Comment:
If it's a singleton, shouldn't the constructor be private so you can only use the method for instantiating the class? 
```suggestion
  private constructor(scope: Construct, id: string) {
```

Reply:
True, I'm just asking myself if we can use `#` instead of `private`

Reply:
we can not :) but fix pushed